src/commands/consolidate.py

```diff
@@ -0,0 +1,330 @@
+    help='Absolute path to the hot wallet password file. '
+    'Default is the file generated with "create-wallet" command.',
+)
+@click.option(
+    '--hot-wallet-file',
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+    envvar='HOT_WALLET_FILE',
+    help='Absolute path to the hot wallet. '
+    'Default is the file generated with "create-wallet" command.',
+)
+@click.option(
+    '-v',
+    '--verbose',
+    help='Enable debug mode. Default is false.',
+    envvar='VERBOSE',
+    is_flag=True,
+)
+@click.option(
+    '--no-confirm',
+    is_flag=True,
+    default=False,
+    help='Skips confirmation messages when provided.',
+)
+@click.option(
+    '--log-level',
+    type=click.Choice(
+        LOG_LEVELS,
+        case_sensitive=False,
+    ),
+    default='INFO',
+    envvar='LOG_LEVEL',
+    help='The log level.',
+)
+@click.command(help='Performs a vault validators consolidation after pectra upgrade.')
+# pylint: disable-next=too-many-arguments,too-many-locals
+def consolidate(
+    network: str,
+    vault: ChecksumAddress,
+    execution_endpoints: str,
+    consensus_endpoints: str,
+    data_dir: str,
+    hot_wallet_file: str | None,
+    hot_wallet_password_file: str | None,
+    verbose: bool,
+    no_confirm: bool,
+    log_level: str,
+    count: int | None = None,
+    from_keys: list[HexStr] | None = None,
+    to_key: HexStr | None = None,
+) -> None:
+    operator_config = OperatorConfig(Path(data_dir))
+    operator_config.load(network=network)
+
+    settings.set(
+        vaults=[vault],
+        network=operator_config.network,
+        data_dir=operator_config.data_dir,
+        execution_endpoints=execution_endpoints,
+        consensus_endpoints=consensus_endpoints,
+        hot_wallet_file=hot_wallet_file,
+        hot_wallet_password_file=hot_wallet_password_file,
+        verbose=verbose,
+        log_level=log_level,
+    )
+    try:
+        asyncio.run(
+            main(
+                vault_address=vault,
+                count=count,
+                from_keys=from_keys,
+                to_key=to_key,
+                no_confirm=no_confirm,
+            )
+        )
+    except Exception as e:
+        log_verbose(e)
+        sys.exit(1)
+
+
+async def main(
+    vault_address: ChecksumAddress,
+    count: int | None,
+    from_keys: list[HexStr] | None,
+    to_key: HexStr | None,
+    no_confirm: bool,
+) -> None:
+    setup_logging()
+    await setup_clients()
+    await _check_validators_manager()
+    chain_head = await get_chain_finalized_head()
+
+    if from_keys is not None and to_key is not None:
+        target_source_public_keys = await _get_selected_target_source_public_keys(
+            from_keys=from_keys,
+            to_key=to_key,
+        )
+
+    else:
+        target_source_public_keys = await _get_all_target_source_public_keys(
+            vault_address=vault_address,
+            block_number=chain_head.block_number,
+            count=count,
+        )
```

Comment:
I suggest we don't allow automatic consolidation, lets keep it simple and require to provide from/to public keys.